src/framework/extensions/internal/legacy/extpluginsloader.cpp

```diff
@@ -31,6 +31,34 @@ using namespace muse;
 using namespace muse::extensions;
 using namespace muse::extensions::legacy;
 
+static Uri makeUri(const io::path_t& rootPath, const io::path_t& path)
+{
+    return Uri("muse://extensions/v1" + path.toQString().sliced(rootPath.size()).toLower().toStdString());
+}
+
+std::map<std::string /*codeKey*/, Uri> ExtPluginsLoader::loadCodekeyUriMap(const io::path_t& defPath, const io::path_t& extPath) const
+{
+    std::map<std::string /*codeKey*/, Uri> data;
+
+    auto loadUris = [this](std::map<std::string /*codeKey*/, Uri>& data, const io::path_t& rootPath) {
+        io::paths_t paths = qmlsPaths(rootPath);
+        for (const io::path_t& path : paths) {
+            std::string codeKey = io::completeBasename(path).toStdString();
+            Uri uri = makeUri(rootPath, path);
+            data.insert({ codeKey, uri });
+        }
+    };
+
+    loadUris(data, defPath);
+    loadUris(data, extPath);
+
+    //! NOTE These plugins have been ported to extensions
+    data["colornotes"] = Uri("musescore://extensions/colornotes");
+    data["addCourtesyAccidentals"] = Uri("musescore://extensions/courtesy_accidentals");
```

Comment:
I also don't fully understand what's happening here

Reply:
For old plugins, the uri is formed according to some rules, there is a static prefix and the name of the plugin qml file is involved
in the old system, plugins were identified by codekey (which is basically just a file name)
to use old settings i need codekey and uri mapping

but I ported some old plugins to the new system and gave them specific uris.